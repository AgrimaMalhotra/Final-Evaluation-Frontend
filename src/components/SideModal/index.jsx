import * as React from 'react';
import './SideModal.css';
import PropTypes from 'prop-types';
import { ADD_COLLECTION_ENTRY_BY_ID } from '../../constants/apiEndPoints';
import { useNavigate } from 'react-router-dom';
import {makeRequest} from '../../utils/makeRequest/makeRequest';

export default function SideModal({ setIsOpen ,inputOptions, collectionDetails}) {
  const [addType, setAddType] = React.useState({});
  const navigate = useNavigate();
  const handleClick = async() => {
    const token = localStorage.getItem('auth_token');
    console.log(collectionDetails);
    await makeRequest(ADD_COLLECTION_ENTRY_BY_ID(collectionDetails.id),navigate,token,{data: {...addType}} );
    setIsOpen(false); };

  return (
    <div>
      <div className="dark-bg" onClick={() => setIsOpen(false)} />
      <div className="right-modal">
        <div className="side-modal">
          <div className="modal-header">
            <h5 className='heading'>{`New ${collectionDetails.name}`}</h5>
          </div>
          <div >
            {inputOptions.map((inputOption,idx) => (
              <div key={idx}>
                <p className="modal-content" >{inputOption}</p>
                <input type="text" className="modal-input" onChange={(e) => setAddType({...addType, inputOption:e.target.value})}/>
              </div>
            ))}
          </div>
          <div className="modal-actions-side">
            <div className="actions-container-side-modal">
              <button className="cancel-button" onClick={handleClick}>Cancel</button>
              <button className="create-button" onClick={handleClick}>Add</button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

SideModal.propTypes = {
  setIsOpen: PropTypes.func.isRequired,
  inputOptions: PropTypes.array.isRequired,
  collectionDetails: PropTypes.object.isRequired,
};