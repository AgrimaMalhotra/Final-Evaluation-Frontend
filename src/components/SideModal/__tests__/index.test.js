import React from 'react';
import {render} from '@testing-library/react';
import SideModal from '../index';

jest.mock('react-router-dom', () => ({ useNavigate: () => jest.fn() }));
describe('SideModal', () => {
  const mockInputOptions = ['name','id'];
  const mockCollectionDetails = {
    'id': 1,
    'name': 'collectionName',
  };
  

  it('should render correctly', () => {
    const component = render(<SideModal setIsOpen={false} inputOptions={mockInputOptions} collectionDetails={mockCollectionDetails}/>);
    expect(component).toMatchSnapshot();
  });
});
