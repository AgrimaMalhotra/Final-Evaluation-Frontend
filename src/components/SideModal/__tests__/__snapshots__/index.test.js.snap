// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideModal should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="dark-bg"
        />
        <div
          class="right-modal"
        >
          <div
            class="side-modal"
          >
            <div
              class="modal-header"
            >
              <h5
                class="heading"
              >
                New collectionName
              </h5>
            </div>
            <div>
              <div>
                <p
                  class="modal-content"
                >
                  name
                </p>
                <input
                  class="modal-input"
                  type="text"
                />
              </div>
              <div>
                <p
                  class="modal-content"
                >
                  id
                </p>
                <input
                  class="modal-input"
                  type="text"
                />
              </div>
            </div>
            <div
              class="modal-actions-side"
            >
              <div
                class="actions-container-side-modal"
              >
                <button
                  class="cancel-button"
                >
                  Cancel
                </button>
                <button
                  class="create-button"
                >
                  Add
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="dark-bg"
      />
      <div
        class="right-modal"
      >
        <div
          class="side-modal"
        >
          <div
            class="modal-header"
          >
            <h5
              class="heading"
            >
              New collectionName
            </h5>
          </div>
          <div>
            <div>
              <p
                class="modal-content"
              >
                name
              </p>
              <input
                class="modal-input"
                type="text"
              />
            </div>
            <div>
              <p
                class="modal-content"
              >
                id
              </p>
              <input
                class="modal-input"
                type="text"
              />
            </div>
          </div>
          <div
            class="modal-actions-side"
          >
            <div
              class="actions-container-side-modal"
            >
              <button
                class="cancel-button"
              >
                Cancel
              </button>
              <button
                class="create-button"
              >
                Add
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
